doc/languages-frameworks/qt.section.md

```diff
@@ -25,12 +25,14 @@ stdenv.mkDerivation {
 
 The same goes for Qt 5 where libraries and tools are under `libsForQt5`.
 
-Any Qt package should include `wrapQtAppsHook` in `nativeBuildInputs`, or explicitly set `dontWrapQtApps` to bypass generating the wrappers.
+Any Qt package should include `wrapQtAppsHook` (or `wrapQtAppsNoGuiHook`) in `nativeBuildInputs`, or explicitly set `dontWrapQtApps` to bypass generating the wrappers.
 
 ::: {.note}
-Qt 6 graphical applications should also include `qtwayland` in `buildInputs` on Linux (but not on platforms e.g. Darwin, where `qtwayland` is not available), to ensure the Wayland platform plugin is available.
 
-This may become default in the future, see [NixOS/nixpkgs#269674](https://github.com/NixOS/nixpkgs/pull/269674).
+`wrapQtAppsHook` will propagate `qtwayland` to ensure the Wayland platform plugin is available. This is useful for graphical applications. On platforms that do not support qtwayland (e.g., Darwin), only `qtbase` will be propagated.
```

Comment:
`qtbase` wasn’t propagated before, right? What benefit do we get from propagating it now?

Reply:
I believe this was to bring back parity with qt5:
https://github.com/NixOS/nixpkgs/blob/1410c766c86d24981d820b942b0ab26876a30c3b/pkgs/development/libraries/qt-5/5.15/default.nix#L360-L364

Reply:
Then I guess I have the same question about Qt 5 😅

Reply:
Don't know honestly, I actually dislike the gist of this PR bc propagating deps is just cementing the cross-comp problems we have.

I'm just keeping this out there hoping a better idea comes through, lol

Reply:
What I don't get is how gtk manages to get away with it, something's just really wrong with qt

Reply:
My understanding is that Wayland support is some kind of dynamically‐loaded platform plugin thing for Qt, which I don’t think GTK does. No idea how X11 support is bundled in to Qt though, and if we could make Wayland get bundled in the same way or something.

Reply:
It's the same for both, `xcb` vs `wayland`
https://doc.qt.io/qt-6/qpa.html#qpa-plugins

Reply:
I think the `qxcb` thing is like, bundled in to the base libraries, though? I’m not entirely sure. Gotcha re: GTK.

Reply:
> and what I meant about gtk is, they propagate gtk3 and somehow manage to not-break cross, which is what I'm planning to look into

Could you explain why propagated deps would stand in conflict with cross?

Reply:
Seems to be picking deps from the wrong arch

Reply:
Okay, I realised why it's propagating `qt5.qtbase.dev`, it's for `qmake`. But the way they propagated it, the wrong arch's setup hook gets chosen and then the build fails.

Reply:
Ah, sorry, I see that’s been done (was working off the PR title which hasn’t been updated).

Reply:
> Okay, I realised why it's propagating `qt5.qtbase.dev`, it's for `qmake`. But the way they propagated it, the wrong arch's setup hook gets chosen and then the build fails (during cross).

I too don't understand why do you want to propagate `qtbase.dev` and `qtwayland.dev`... and how is that related to qmake...

Reply:
> why do you want to propagate `qtbase.dev` and `qtwayland.dev`

_I_ do not, this is existing code.